pandas/tests/indexes/test_base.py

```diff
@@ -2359,12 +2359,14 @@ def test_intersection_base(self, sort):
         if PY3 and sort:
             # unorderable types
             warn_type = RuntimeWarning
+            expected = Index([0, 'a', 1])
```

Comment:
At the moment ``Index.intersection`` does not use ``sorting.safe_sort`` which tries to sort arrays of mixed type (int and str) by ordering int before str in Python 3. I could try to change this either in this PR or in a separate one if this is getting too big already.

Reply:
Decided to use ``sort_safe`` in ``Index.intersection`` in this PR in the next commit